### PR TITLE
add log time in sharedcpp

### PR DESCRIPTION
**Pull request by ihkao**

### CODE REVIEW

File: rmw_fastrtps_shared_cpp/src/rmw_take.cpp [hunk 1]
```diff
@@ -101,6 +103,7 @@ _take(
       if (sinfo.valid_data) {
         if (message_info) {
           _assign_message_info(identifier, message_info, &sinfo);
+          fabric_functions::fabric_logger(message_info, subscription, "FastRTPS");
```
Comment (JWhitleyWork) [inline rmw_fastrtps_shared_cpp/src/rmw_take.cpp:106]: This is inside `take` whereas with eCAL, it's in `take_with_info`. We should definitely have them at least in the same function.

**Comment by JWhitleyWork** [comment]
Also, thoughts on just using the header file (with definition)?

File: rmw_fastrtps_shared_cpp/src/rmw_take.cpp (hunk 1 above)
Comment (ihkao) [reply rmw_fastrtps_shared_cpp/src/rmw_take.cpp:106]: Unfortunately, we can't get the `message_info->source_timestamp` from `take_with_info` in FastRTPS.  
message_info is assigned by the line which is : 
`_assign_message_info(identifier, message_info, &sinfo)`  in `_take`.
And `take_with_info` calls `_take`. Therefore, message_info is empty in `take_with_info`.

**Comment by ihkao** [comment]
> Also, thoughts on just using the header file (with definition)?

Fix in [11b7ea8](https://github.com/upowerrobotics/rmw_fastrtps/pull/1/commits/11b7ea84e8028c38adb886467ebaba57abc3d32e)

File: rmw_fastrtps_shared_cpp/src/rmw_take.cpp (hunk 1 above)
Comment (JWhitleyWork) [reply rmw_fastrtps_shared_cpp/src/rmw_take.cpp:106]: Thanks for the info!

**Review by JWhitleyWork — APPROVED** [review]
LGTM

**Pull request merged**